pkg/cloud/openstack/cluster/actuator.go

```diff
@@ -87,20 +87,41 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster) error {
 		return err
 	}
 
+	secGroupService, err := clients.NewSecGroupService(client)
+	if err != nil {
+		return err
+	}
+
 	// Load provider config.
 	_, err = providerv1.ClusterSpecFromProviderSpec(cluster.Spec.ProviderSpec)
 	if err != nil {
 		return errors.Errorf("failed to load cluster provider config: %v", err)
 	}
 
 	// Load provider status.
-	_, err = providerv1.ClusterStatusFromProviderStatus(cluster.Status.ProviderStatus)
+	providerStatus, err := providerv1.ClusterStatusFromProviderStatus(cluster.Status.ProviderStatus)
 	if err != nil {
 		return errors.Errorf("failed to load cluster provider status: %v", err)
 	}
 
 	// Delete other things
 
+	if providerStatus.GlobalSecurityGroup != nil {
```

Comment:
Do we need to check `managedSecurityGroups` to see if it is enabled or not when deleting the security groups?

Reply:
I suppose checking `managedSecurityGroups` parameter here is not really necessary, because `providerStatus.GlobalSecurityGroup` and `providerStatus.ControlPlaneSecurityGroup` values provide us information about security groups being created (this fields are filling up only if `managedSecurityGroups` is true). If such parameters are not `nil`, then security groups were created and we need to delete them.

Reply:
make sense, thanks @eromanova 

/approve

/cc @jichenjc @chrigl